hub/src/server/authentication.js

```diff
@@ -0,0 +1,191 @@
+/* @flow */
+
+import bitcoin from 'bitcoinjs-lib'
+import crypto from 'crypto'
+import { decodeToken, TokenSigner, TokenVerifier } from 'jsontokens'
+import { ecPairToHexString } from 'blockstack'
+import { ValidationError } from './errors'
+import logger from 'winston'
+
+const DEFAULT_STORAGE_URL = 'storage.blockstack.org'
+export const LATEST_AUTH_VERSION = 'v1'
+
+function pubkeyHexToECPair (pubkeyHex) {
+  const pkBuff = Buffer.from(pubkeyHex, 'hex')
+  return bitcoin.ECPair.fromPublicKeyBuffer(pkBuff)
+}
+
+export class V1Authentication {
+  constructor(token: string) {
+    this.token = token
+  }
+
+  static fromAuthPart(authPart: string) {
+    if (!authPart.startsWith('v1:')) {
+      throw new ValidationError('Authorization header should start with v1:')
+    }
+    const token = authPart.slice('v1:'.length)
+    const decodedToken = decodeToken(token)
```

Comment:
Consider surrounding this with a try/catch block that throws ValidationError(), since the token could be malformed.